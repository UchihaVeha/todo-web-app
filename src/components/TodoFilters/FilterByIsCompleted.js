// @flow
import React from 'react';
import { pick } from 'lodash';
import { pure, compose, withHandlers } from 'recompose';
import Radio, { RadioGroup } from 'material-ui/Radio';
import { FormControlLabel } from 'material-ui/Form';
import { withStyles } from 'material-ui/styles';
import type { isCompletedFilterProps } from 'modules/todo/reducers';

const styles = (theme: Object) => ({
  root: {
    marginBottom: 18
  },
  label: {
    color: theme.palette.common.darkWhite
  },
  checked: {
    color: theme.palette.secondary.A200
  }
});

type Props = {
  value: isCompletedFilterProps,
  onChangeHandler: () => void,
  classes: Object
};

export const FilterByIsCompletedComponent = ({
  value,
  onChangeHandler,
  classes
}: Props) => (
  <RadioGroup
    row
    value={value}
    className={classes.root}
    onChange={onChangeHandler}
  >
    <FormControlLabel
      value="all"
      control={
        <Radio className={classes.label} checkedClassName={classes.checked} />
      }
      label="All"
      classes={pick(classes, 'label')}
    />
    <FormControlLabel
      value="completed"
      control={
        <Radio className={classes.label} checkedClassName={classes.checked} />
      }
      label="Completed"
      classes={pick(classes, 'label')}
    />
    <FormControlLabel
      value="notCompleted"
      control={
        <Radio className={classes.label} checkedClassName={classes.checked} />
      }
      label="Not completed"
      classes={pick(classes, 'label')}
    />
  </RadioGroup>
);

export default compose(
  pure,
  withHandlers({
    onChangeHandler: props => (e, value) => {
      props.onChange(value);
    }
  }),
  withStyles(styles, { name: 'FilterByIsCompleted' })
)(FilterByIsCompletedComponent);
