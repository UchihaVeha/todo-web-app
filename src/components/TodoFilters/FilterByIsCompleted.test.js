import React from 'react';
import { shallow } from 'enzyme';
import FilterByIsCompleted from './FilterByIsCompleted';

describe('TodoFilters/FilterByIsCompleted component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<FilterByIsCompleted />);
  });

  it('should render pure', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render withHandlers', () => {
    expect(wrapper.dive()).toMatchSnapshot();
  });

  it('should render withStyles', () => {
    expect(wrapper.dive().dive()).toMatchSnapshot();
  });

  it('should render component', () => {
    expect(
      wrapper
        .dive()
        .dive()
        .dive()
    ).toMatchSnapshot();
  });
});
