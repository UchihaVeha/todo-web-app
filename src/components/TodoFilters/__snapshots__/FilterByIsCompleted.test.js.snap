// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoFilters/FilterByIsCompleted component should render component 1`] = `
<RadioGroup
  className="FilterByIsCompleted-root-1"
  onChange={[Function]}
  row={true}
>
  <withStyles(FormControlLabel)
    classes={
      Object {
        "label": "FilterByIsCompleted-label-2",
      }
    }
    control={
      <Radio
        checkedClassName="FilterByIsCompleted-checked-3"
        className="FilterByIsCompleted-label-2"
      />
    }
    label="All"
    value="all"
  />
  <withStyles(FormControlLabel)
    classes={
      Object {
        "label": "FilterByIsCompleted-label-2",
      }
    }
    control={
      <Radio
        checkedClassName="FilterByIsCompleted-checked-3"
        className="FilterByIsCompleted-label-2"
      />
    }
    label="Completed"
    value="completed"
  />
  <withStyles(FormControlLabel)
    classes={
      Object {
        "label": "FilterByIsCompleted-label-2",
      }
    }
    control={
      <Radio
        checkedClassName="FilterByIsCompleted-checked-3"
        className="FilterByIsCompleted-label-2"
      />
    }
    label="Not completed"
    value="notCompleted"
  />
</RadioGroup>
`;

exports[`TodoFilters/FilterByIsCompleted component should render pure 1`] = `<withHandlers(withStyles(FilterByIsCompletedComponent)) />`;

exports[`TodoFilters/FilterByIsCompleted component should render withHandlers 1`] = `
<withStyles(FilterByIsCompletedComponent)
  onChangeHandler={[Function]}
/>
`;

exports[`TodoFilters/FilterByIsCompleted component should render withStyles 1`] = `
<FilterByIsCompletedComponent
  classes={
    Object {
      "checked": "FilterByIsCompleted-checked-3",
      "label": "FilterByIsCompleted-label-2",
      "root": "FilterByIsCompleted-root-1",
    }
  }
  onChangeHandler={[Function]}
/>
`;
